src/__tests__/auth.spec.js

```diff
@@ -1,5 +1,152 @@
-describe('Main test suite', () => {
-  it('App Test', () => {
-    expect(true).toEqual(true);
+// eslint-disable-next-line node/no-unpublished-require
+const request = require('supertest');
+const app = require('../app');
+const connectDatabase = require('../db/connection');
+
+beforeAll(async () => {
+  await connectDatabase();
+});
+
+const validUserExample = {
+  email: 'niloaydin@domain.com.tr',
+  password: 'Nilay-123',
+};
+
+let jwtToken = null;
+describe('AUTH TESTS', () => {
+  describe('POST /api/auth/login', () => {
+    it('should return a token in a cookie and success message ', (done) => {
+      request(app)
+        .post('/api/auth/login')
+        .set('Content-Type', 'application/json')
+        .send({
+          email: validUserExample.email,
+          password: validUserExample.password,
+        })
+        .expect('Content-Type', /json/)
+        .expect(201, (err, res) => {
+          if (err) return done(err);
+          expect(res.headers['set-cookie']).toBeDefined();
+          expect(res.headers['set-cookie']).toBeTruthy();
+          expect(res.body.message).toBe('User sucesfully signed in!');
+          // eslint-disable-next-line prefer-destructuring
+          jwtToken = res.headers['set-cookie'][0].split(';')[0];
+          return done();
+        });
+    });
+    it('should return an error when email thats not registered is used', (done) => {
+      request(app)
+        .post('/api/auth/login')
+        .set('Content-Type', 'application/json')
+        .send({
+          email: 'wrongemail@wrongemail.com',
+          password: validUserExample.password,
+        })
+        .expect('Content-Type', /json/)
+        .expect(401, (err, res) => {
+          if (err) return done(err);
+          expect(res.body.message).toBe('Wrong username or password!');
+          return done();
+        });
+    });
+    it('should return an error when password thats not registered is used', (done) => {
+      request(app)
+        .post('/api/auth/login')
+        .set('Content-Type', 'application/json')
+        .send({
+          email: validUserExample.email,
+          password: 'wrongpasswordthaticreatedfortestpurposes',
```

Comment:
i see what you mean by clear naming :)))

Reply:
Because I am testing if the code is gonna return an error if there is a wrong password passed